hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1427,6 +1469,38 @@ int MDMParser::_cbCCID(int type, const char* buf, int len, CStringHelper* str)
     return WAIT;
 }
 
+void MDMParser::_checkVerboseC_reg(void) {
+    if (!_attached) {
+        return;
+    }
+
+    if (_lastVerboseC_regUpdate &&
+        (HAL_Timer_Get_Milli_Seconds() - _lastVerboseC_regUpdate) >= MDM_C_REG_POLL_INTERVAL_MS)
+    {
+        LOCK();
+        if (!_checkModem()) {
+            return;
+        }
+        _lastVerboseC_regUpdate = HAL_Timer_Get_Milli_Seconds();
```

Comment:
The following AT commands `AT+C*REG?` have 60 sec timeout. Might it be good to set this _after_ getting a response from the commands below, so that we will actually wait `MDM_C_REG_POLL_INTERVAL_MS` before the next call?

Reply:
This timer is arbitrarily set to 30 seconds, just to be kind of fast for recovery, but not super duper noisy.  I would not want to set this for the first time after any C*REG response that might never come before a brown out/reset occurs.  This timer can just run asynchronously in the background forever not governed by anything but itself... which will keep it running despite weird edge cases we don't predict now.

Reply:
We also are guaranteeing recovery within 2 minutes from these failures, by way of CELLUAR_03/04 test assertions. We could better ensure that 2 min by making C*REG polling faster if necessary.  I was seeing recovery in about 60 seconds though, so I think we have a decent margin already.

Reply:
I believe I was referring to the edge-case where the response for `AT+CEREG?` takes a long time like 30 or 40 sec, in which case we will be querying them back to back, because `MDM_C_REG_POLL_INTERVAL_MS` would have elapsed by then. Again, it's a weird edge case that you have a valid point not to account for. 👍

Reply:
This _checkModem() with AT/OK check call will be like 99% defense against calling the C*REG? commands with 60 second timeout when the AT interface is dead.  If the AT interface dies right when C*REG is called, it will take 60 seconds... but then the next _checkModem call will cause it to fail completely as two _error's will have accrue quickly with _checkModem, partially because _lastVerboseC_regUpdate is set after that so that it gets called again right away and fails.

I see what you mean now though.  Typically C*REG calls will be fast.  But if they timeout as mentioned above (after an successful AT/OK check) and we have to wait 60s... I wouldn't want to wait another 30s before the double _checkModem call.  It's good to think these different test cases though, so I appreciate the suggestion :D

Reply:
Cool - thanks for helping understand!..